### PR TITLE
[FEATURE] Enable Expectation "expect_column_values_to_be_in_set" to be Self-Initializing

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Create default profiler for the expectation "expect_column_values_to_be_in_set"
* Update utilities and parts of the infrastructure
* Add new integration tests
* Update existing unit tests

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Part of JIRA: GREAT-464/GREAT-498
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/core/expect_column_values_to_be_in_set.py [hunk 1]
```diff
@@ -104,17 +105,62 @@ class ExpectColumnValuesToBeInSet(ColumnMapExpectation):
...
-    default_kwarg_values = {"value_set": None, "parse_strings_as_datetimes": False}
-    args_keys = (
-        "column",
-        "value_set",
+
+    default_profiler_config: RuleBasedProfilerConfig = RuleBasedProfilerConfig(
+        name="expect_column_values_to_be_in_set",  # Convention: use "expectation_type" as profiler name.
+        config_version=1.0,
+        variables={
+            "mostly": 1.0,
```
Comment (anthonyburdi) [inline great_expectations/expectations/core/expect_column_values_to_be_in_set.py:128]: What if we remove the mostly param to keep the config shorter since it is set at 100%?

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py [hunk 2]
```diff
@@ -130,11 +133,11 @@ def _build_parameters(
         unique_parameter_values: Set[
             Any
         ] = _get_unique_values_from_nested_collection_of_sets(
-            parameter_value_node["value"]
+            parameter_value_node.value
         )
 
         return (
-            unique_parameter_values,
+            sorted(unique_parameter_values),
```
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py]: Interesting! TIL you can sort a set.

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py [hunk 3]
```diff
@@ -1440,3 +1440,73 @@ def test_quentin_expect_column_quantile_values_to_be_between_auto_yes_default_pr
...
+    try:
+        # noinspection PyUnusedLocal
+        suite = context.get_expectation_suite(
+            expectation_suite_name=expectation_suite_name
+        )
+    except ge_exceptions.DataContextError:
+        suite = context.create_expectation_suite(
+            expectation_suite_name=expectation_suite_name
+        )
+        print(f'Created ExpectationSuite "{suite.expectation_suite_name}".')
```
Comment (anthonyburdi) [inline tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py:1470]: We should know here whether the suite exists so you can probably get rid of the try/except

File: great_expectations/expectations/core/expect_column_values_to_be_in_set.py [hunk 4]
```diff
@@ -104,17 +105,62 @@ class ExpectColumnValuesToBeInSet(ColumnMapExpectation):
...
+                        "value_set": "$parameter.value_set.value",
+                        "mostly": "$variables.mostly",
+                        "meta": {
+                            "profiler_details": "$parameter.value_set.details",
+                        },
+                    },
+                ],
+            },
+        },
     )
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_be_in_set.py]: Is this reusable at all between different expectations? I'd love to save us from having to write out full configs each time if possible but that may not be realistic.

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py [hunk 5]
```diff
@@ -236,7 +236,7 @@ def _build_binary_list(
 
     def _build_boolean_result(
         self,
-        binary_list: List[Union[bool, str]],
+        binary_list: Union[ParseResults, List[Union[bool, str]]],
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:239]: Why this change? Can we possibly narrow down to a single type?

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py [hunk 6]
```diff
@@ -153,5 +156,11 @@ def _get_unique_values_from_nested_collection_of_sets(
...
 
     flattened: List[Set[Any]] = list(itertools.chain.from_iterable(collection))
-    unique_values: Set[Any] = set().union(*flattened)
+    unique_values: Set[Any] = set(
+        filter(
+            lambda element: element is not None,
+            set().union(*flattened),
+        )
+    )
+
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:173]: This is some beautiful code but a little hard to understand. Can we simplify it at all? Or perhaps add a comment?

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py (hunk 2 above)
Comment (cdkini) [inline great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py]: Why is this necessary?

**Review by cdkini — COMMENTED** [review]
Looks solid! Will ask for other eyes on this since I'm not super well versed in the prior self-initializing expectations PR.

File: tests/rule_based_profiler/parameter_builder/test_value_set_multi_batch_parameter_builder.py [hunk 7]
```diff
@@ -80,7 +80,7 @@ def test_value_set_multi_batch_parameter_builder_alice_single_batch_numeric(
     )
 
     fully_qualified_parameter_name_for_value: str = "$parameter.my_event_type_value_set"
-    expected_value_set: Set[int] = {73, 19, 22}
+    expected_value_set: List[int] = [19, 22, 73]
```
Comment (anthonyburdi) [inline tests/rule_based_profiler/parameter_builder/test_value_set_multi_batch_parameter_builder.py:82]: If possible I'd love to leave this as a set which is more meaningful since this is the set of unique values.

**Review by anthonyburdi — COMMENTED** [review]
Looks great - just a suggestion and question

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py [hunk 8]
```diff
@@ -5,14 +5,13 @@
...
+from great_expectations.rule_based_profiler.types import Domain, ParameterContainer
 from great_expectations.rule_based_profiler.types.parameter_container import (
     ParameterNode,
 )
+from great_expectations.rule_based_profiler.util import (
+    get_parameter_value_and_validate_return_type,
+)
 
 
 class ValueSetMultiBatchParameterBuilder(MetricMultiBatchParameterBuilder):
```
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:17]: As discussed, please add a note that this now filters null values @alexsherstinsky

File: great_expectations/expectations/core/expect_column_values_to_be_in_set.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/core/expect_column_values_to_be_in_set.py:128]: @anthonyburdi This is a placeholder so that it could be changed here, but used generically as `@variable.mostly` elsewhere.

File: tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py (hunk 3 above)
Comment (alexsherstinsky) [reply tests/integration/profiling/rule_based_profilers/test_profiler_user_workflows.py:1470]: @anthonyburdi this is the "safe" pattern from elsewhere (not thinking there is any harm in keeping it like this).

File: tests/rule_based_profiler/parameter_builder/test_value_set_multi_batch_parameter_builder.py (hunk 7 above)
Comment (alexsherstinsky) [reply tests/rule_based_profiler/parameter_builder/test_value_set_multi_batch_parameter_builder.py:82]: @anthonyburdi From our discussion -- "set" is not serializable.

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py (hunk 8 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:17]: @anthonyburdi Done (in next commit).

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py]: @cdkini Getting rid of it (found it to be unnecessary).  Thanks.

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py (hunk 6 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:173]: @cdkini I opted for the comment.  Thanks! :)

File: great_expectations/expectations/core/expect_column_values_to_be_in_set.py (hunk 4 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/core/expect_column_values_to_be_in_set.py]: @cdkini My thinking on reusability is as follows: we create the (relatively simple) default profiler configurations for the individual expectations.  Then, it will be simple to refer them for various uses (in an a-la-carte fashion).  I have a notebook I can show you that is doing this (hat tip to @talagluck ).  Thanks!

File: great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py (hunk 5 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py:239]: @cdkini The previous version was getting PyCharm complaints.  @NathanFarmer if you think of a better way, please make the fix as you deem appropriate.  Thanks.

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py]: @anthonyburdi We got rid of sorting.  Thanks for the discussion.

File: great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py [hunk 9]
```diff
@@ -156,6 +156,12 @@ def _get_unique_values_from_nested_collection_of_sets(
     """
 
     flattened: List[Set[Any]] = list(itertools.chain.from_iterable(collection))
+
+    """
+    In multi-batch data analysis, values can be empty and missin, resulting in "None" added to set.  However, due to
```
Comment (anthonyburdi) [inline great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:163]: @alexsherstinsky Can you also add `This ParameterBuilder filters null values from the unique value_set.` to the class docstring?

Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py:163]: @anthonyburdi Done (committed).  Thanks.

**Review by anthonyburdi — APPROVED** [review]
LGTM!
